rules/prefer-math-min-max.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const MESSAGE_ID = 'prefer-math-min-max';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `{{replacement}}` to simplify ternary expressions.',
+};
+
+/**
+@param {import('eslint').Rule.RuleContext} context
+@param {import('estree').ConditionalExpression} node
+@param {string} method
+*/
+function reportPreferMathMinOrMax(context, node, left, right, method) {
+	const {sourceCode} = context;
+
+	context.report({
+		node,
+		messageId: MESSAGE_ID,
+		data: {
+			replacement: `${method}()`,
+			value: sourceCode.getText(node),
+		},
+		fix: fixer => fixer.replaceText(node, `${method}(${sourceCode.getText(left)}, ${sourceCode.getText(right)})`),
+	});
+}
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => ({
+	/** @param {import('estree').ConditionalExpression} node */
+	ConditionalExpression(node) {
+		const {test, consequent, alternate} = node;
+
+		if (test.type !== 'BinaryExpression') {
+			return;
+		}
+
+		const {sourceCode} = context;
+		const {operator, left, right} = test;
+
+		const leftCode = sourceCode.getText(left);
+		const rightCode = sourceCode.getText(right);
+		const alternateCode = sourceCode.getText(alternate);
+		const consequentCode = sourceCode.getText(consequent);
+
+		switch (operator) {
```

Comment:
I believe this can be simplify to just two branch, can you try?

Reply:
dc5aedcb0e35ad6345541fd7ccbb7f19d173bb97

Reply:
There are still 4 branches, the if condition can be merged.

Reply:
These four branches are definitely necessary and cannot be eliminated

Reply:
Updated in 9bdef777c5997c9bedf92e38fe139434adf9b57d